File: python/23.06/599.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        dic = dict()
        for i in range(len(list1)):
            dic[list1[i]] = i
        
        ans = []
        ansi = float('inf')

        for i in range(len(list2)):
            if dic.get(list2[i]) != None:
                if dic[list2[i]] + i < ansi:
                    ansi = dic[list2[i]] + i 
                    ans = [list2[i]]
                elif dic[list2[i]] + i == ansi:
                    ans.append(list2[i])
        
        return ans

File: javascript/23.06/599.js
/**
 * @param {string[]} list1
 * @param {string[]} list2
 * @return {string[]}
 */
var findRestaurant = function(list1, list2) {
    let cnt = 10 ** 9
    let ans = []
    for (let i in list1) {
        for (let j in list2) {
            if (list1[i] === list2[j]) {
                if (cnt > Number(i) + Number(j)) {
                    cnt = Number(i) + Number(j)
                    ans = [list1[Number(i)]]
                } else if (cnt === Number(i) + Number(j)) {
                    ans.push(list1[i])
                }
            }
        }
    }

    return ans
};
